### DISCUSSION TITLE
GRPC server reflection protocol

### DISCUSSION THREAD

**Discussion by Bouwdie** [Kong/insomnia#3942] [Ideas] [answered]
Hello everyone. I am very happy with the coming of GRPC into the insomnia tooling. I think it is an opportunity to get ahead of the game in API development and testing - one tool to rule them all (whilst keeping it simple). One thing that might be an interesting feature is an implementation of the GRPC reflection protocol: https://github.com/grpc/grpc/blob/master/doc/server-reflection.md It would make connecting to services that much easier since you do not have to acquire the proto files. Which for a simple service is a nuisance but doable, for a more complex one using the buf toolset with external dependencies it gets more complex. Personally (and in my company) we implement the reflection protocol for all our services to lower the boundaries of starting with GRPC. I think it would not be difficult to add, and an easy step to make GRPC more accessible. What are your thoughts? :1234:

**Comment by dimitropoulos** [comment]
I'm very excited to hear ideas like this!  Thanks!

The first thing I'd like to know is: could you assist in getting a testing setup going so we can make a server that we can test implementing this with?

**Comment by wdawson** [comment, marked as answer]
Hey folks, it looks like this is a duplicate of #3669, I'm going to lock this conversation and copy your comments there. Please feel free to upvote that one if you still would like to see this in Insomnia 🚀
